### PR TITLE
[BugFix] avoid translate into cross join when conjunct is not empty

### LINKED ISSUE #10996: Left join without join conjuncts should not be transformed to cross join
### Steps to reproduce the behavior (Required)

```sql
select * from t0 left join t1 on t1.id  = 2;
```

- The correct plan should be a `LEFT JOIN` with `t1.id = 2` push down
- But right now is a `CROSS JOIN`

### StarRocks version (Required)
- 2.4-rc0

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] bugfix
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #10996

## Problem Summary(Required) ：

An outer join should not be translated into cross join when conjuncts is not empty.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/JoinTest.java [hunk 1]
```diff
@@ -105,7 +105,7 @@ public void testCrossJoin() throws Exception {
         plan = getFragmentPlan(sql);
         Assert.assertTrue(plan,
                 plan.contains("  3:NESTLOOP JOIN\n" +
-                        "  |  join op: CROSS JOIN\n" +
+                        "  |  join op: INNER JOIN\n" +
```
Comment (kangkaisen) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/JoinTest.java:108]: which this could be inner join?

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownJoinPredicateBase.java|1|1|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**